packages/docsearch-react/src/DocSearch.tsx

```diff
@@ -42,24 +42,36 @@ export interface DocSearchProps {
   initialQuery?: string;
   navigator?: AutocompleteOptions<InternalDocSearchHit>['navigator'];
   translations?: DocSearchTranslations;
-  getMissingResultsUrl?: ({ query: string }) => string;
+  getMissingResultsUrl?: ({ query }: { query: string }) => string;
 }
 
 export function DocSearch(props: DocSearchProps) {
   const searchButtonRef = React.useRef<HTMLButtonElement>(null);
+  const activeElementRef = React.useRef<Element>(document.body);
   const [isOpen, setIsOpen] = React.useState(false);
   const [initialQuery, setInitialQuery] = React.useState<string | undefined>(
     props?.initialQuery || undefined
   );
 
   const onOpen = React.useCallback(() => {
     setIsOpen(true);
+    activeElementRef.current = document.activeElement || document.body;
   }, [setIsOpen]);
 
   const onClose = React.useCallback(() => {
     setIsOpen(false);
   }, [setIsOpen]);
 
+  useEffect(() => {
+    if (!isOpen) {
+      if (activeElementRef.current) {
+        if (activeElementRef.current instanceof HTMLElement) {
```

Comment:
not a big deal but we can wrap it in a single if statement

Reply:
done.